homeassistant/helpers/entity_component.py

```diff
@@ -263,6 +263,8 @@ def _async_update_group(self):
             self.logger.error(err)
             return None
 
+        await conf_util.async_hass_config_merge(self.hass, conf)
```

Comment:
Makes more sense if this is done inside of `async_hass_config_yaml`, the entity component should not be aware that any merging is happening.

Reply:
Yeah, does make sense. I moved the call.
However since both `bootstrap` and `script/check_config` use `load_yaml_config_file` we still need the merge call there.